woocommerce/woocommerce-gutenberg-products-block
# Product Collection: Hide filters in Inspector Controls depending on chosen Collection

## What

Hide some Product Collection filters in Inspector Controls based on the chosen Collection.

Note: This PR is created based on https://github.com/woocommerce/woocommerce-blocks/pull/10952 and is not targeting `trunk` but `add/10822-new-flow-for-product-collection`

Fixes https://github.com/woocommerce/woocommerce/issues/42087

## Why

Let's take New Arrivals as an example. What defines New Arrivals is the product order: from newest to oldest. Users can apply additional filters on top of it, for example, "On Sale" but shouldn't be able to change ordering because that would no longer be New Arrivals Collection.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Go to Editor
2. Insert Product Collection block
3. Click "Choose Collection"
4. Choose "Product Catalog"
5. Expected: There's "Inherit query from template" available and it's enabled by default
6. Focus on Product Collection and click "Choose collection" in toolbar
 
<img width="936" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/20098064/98153134-80b9-4886-8729-cb4ca282fbc7">

7. Choose "New Arrivals"
8. Expected: "Order By" option is NOT available in Inspector Controls
9. Repeat 6 - 8 for other Collections:
  - Collection: "Featured", Expected: "Featured" filter is NOT available in Inspector Controls
  - Collection: "Top Rated", Expected: "Order By" option is NOT available in Inspector Controls
  -  Collection: "On Sale", Expected: "On Sale" filter is NOT available in Inspector Controls
  -  Collection: "Best Sellers", Expected:"Order By" option is NOT available in Inspector Controls
  -  Collection: "New Arrivals", Expected: "Featured" filter is NOT available in Inspector Controls


## Screenshots or screencast



| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility


Required:


## Checklist

Required:


Conditional:


## Changelog


> Product Collection: Hide Collection implementation in UI by hiding specific filters and options in Inspector Controls
